### ISSUE TITLE
Help Need

### ISSUE THREAD

**Issue by vaithitheone** [cowinapi/developer.cowin#263]
Can we use this api to get vaccination report? [this](https://cdn-api.co-vin.in/api/v1/reports/v2/getPublicReports?state_id=&district_id=&date=2021-05-18).

**Comment by anandvijay094** [comment]
Yes you can use the public Certificate api 
( Here I'm using Postman )

Step 1: call generateOTP  api  and pass mobile no. In request to generate otp & it will send txnid in response.

Step 2 : call confirmOTP api & pass txid & otp using sha256 received in 1st step & it will generate a token as response.

Step 3 : call Certificate api & pass the token received in the 2nd step in  bearer token available in authorization option in postman & pass beneficiary id  in the request body. Then it will send you the binary data of the vaccination report in response. In postman you will see the option " Save response"  so here you will get the option  "save to file" , this will print the report into pdf.

Note : OTP is valid for 3 minutes so try to generate a token within 3 minutes .The token is valid for almost 10-15 minutes if you are using it continuosly. 

Link of certificate api : 
https://cdn-api.co-vin.in/api/v2/registration/certificate/public/download?beneficiary_reference_id=1234567890123 

Any suggestions are welcome.

**Comment by vaithitheone** [comment]
yes i know that , I'm not asking about that can you check that api link? And also in this api i can't test it because i'm not sure about output format .
This one :  https://cdn-api.co-vin.in/api/v1/reports/v2/getPublicReports?state_id=&district_id=&date=2021-05-18. Not listed in the api directory So only posted here.

**Comment by kichappa** [comment]
@wifirevaithi You can use it as you intend. To use it, parse the response content as a `json` type. This will let you use it as a dictionary.  You can do this in python like so,
```
import requests

URL = "https://cdn-api.co-vin.in/api/v1/reports/v2/getPublicReports?state_id=&district_id=&date=2021-05-18"
response = requests.get(URL, headers={
    "accept": "application/json",
    "Accept-Language": "en_US",
    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36",
    "authorization": "Bearer {}".format(bearer_token)
    })

# parsing the response as a json
print(response.json())
```

**Comment by Aarushi-1** [comment]
> Yes you can use the public Certificate api
> ( Here I'm using Postman )
> 
> Step 1: call generateOTP api and pass mobile no. In request to generate otp & it will send txnid in response.
> 
> Step 2 : call confirmOTP api & pass txid & otp using sha256 received in 1st step & it will generate a token as response.
> 
> Step 3 : call Certificate api & pass the token received in the 2nd step in bearer token available in authorization option in postman & pass beneficiary id in the request body. Then it will send you the binary data of the vaccination report in response. In postman you will see the option " Save response" so here you will get the option "save to file" , this will print the report into pdf.
> 
> Note : OTP is valid for 3 minutes so try to generate a token within 3 minutes .The token is valid for almost 10-15 minutes if you are using it continuosly.
> 
> Link of certificate api :
> https://cdn-api.co-vin.in/api/v2/registration/certificate/public/download?beneficiary_reference_id=1234567890123
> 
> Any suggestions are welcome.

where to get the beneficiary_reference_id ?

**Comment by jeet2208** [comment]
From Arogya Setu App  or Cowin website . you can get the beneficiary_reference_id. Just click on any hyperlink of Person name , who has taken at least first dose.
